Automattic/wp-calypso
# Plans 2023: Pass selectedSiteId as a prop to pricing meta hook

Related to https://github.com/Automattic/wp-calypso/pull/85359 (reverted), https://github.com/Automattic/wp-calypso/pull/85486 (take 2)

## Proposed Changes

- Uncouples the pricing meta hook further from Calypso dependencies by passing in `selectedSiteId`. https://github.com/Automattic/wp-calypso/pull/85486 migrates the pricing, the hook itself remains linked to other imports.
- Given that this hook's use propagated to several places, then the changes have a wider impact, so need thorough testing
- Some obsolete uses of `siteId` in the plans-grid components have been cleaned up too, and all remaining cases are of `selectedSiteId` (it conveys better semantics off the bat)

## Testing Instructions



- Typechecks/unit pass
- Check pricing in `/plans` and `/start/plans`
- Confirm savings when clicking "monthly" toggle in `/start/plans` are correct (55%)
- Confirm that a purchased plan is rendered with the price it was purchased with in the Spotlight card (assign an older / 2021 personal USD plan through SA to a site)
- Confirm `/plans/[fresh woo site]` renders intro offers (get a site from `/setup/wooexpress`
- Confirm actions are correct: 
  - can purchase in `/start/plans`
  - cannot downgrade in `/plans/[business]`
  - etc.

## Pre-merge Checklist
